ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -469,6 +469,9 @@ public Transaction getTransactionById(Object id) {
     }
 
     public Object getCurrentTransactionId() {
+        if (com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.getTransaction() == null)
```

Comment:
Similar comment,  TransactionImple.getTransaction() can do a lot of work so I think you need to cache the result in a local variable in order to avoid the second call to getTransaction.